content/en/docs/components/katib/getting-started.md

```diff
@@ -0,0 +1,164 @@
++++
+title = "Getting Started"
+description = "Get started with Katib"
+weight = 30
+
+mathjax = true
++++
+
+This guide describes how to get started with Katib and run a few examples.
+
+## Prerequisites
+
+You need to install the following Katib components to run examples:
+
+- Katib [installed](/docs/components/katib/installation/#installing-katib).
+- Katib Python SDK [installed](/docs/components/katib/installation/#installing-katib-python-sdk).
+
+## Getting Started with Katib Python SDK
+
+You can run your first hyperparameter tuning Katib Experiment using Python SDK.
+
+In the following example we are going to maximize a simple objective function:
+
+<p>
+$$
+F(a,b) = 4a - b^2
+$$
+</p>
+
+The bigger \(a\) and the lesser \(b\) value, the bigger the function value \(F\).
+
+```python
+# [1] Create an objective function.
+def objective(parameters):
+    # Import required packages.
+    import time
+    time.sleep(5)
+    # Calculate objective function.
+    result = 4 * int(parameters["a"]) - float(parameters["b"]) ** 2
+    # Katib parses metrics in this format: <metric-name>=<metric-value>.
+    print(f"result={result}")
+
+import kubeflow.katib as katib
+
+# [2] Create hyperparameter search space.
+parameters = {
+    "a": katib.search.int(min=10, max=20),
+    "b": katib.search.double(min=0.1, max=0.2)
+}
+
+# [3] Create Katib Experiment with 12 Trials and 2 CPUs per Trial.
+name = "tune-experiment"
+katib.KatibClient().tune(
+    name=name,
+    objective=objective,
+    parameters=parameters,
+    objective_metric_name="result",
+    max_trial_count=12,
+    resources_per_trial={"cpu": "2"},
+)
+
+# [4] Wait until Katib Experiment is complete
+katib.KatibClient().wait_for_experiment_condition(name=name)
+
+# [5] Get the best hyperparameters.
+print(katib.KatibClient().get_optimal_hyperparameters(name))
+```
+
+## Getting Started with Katib Experiment YAML
```

Comment:
I can move this to a separate guide. I think, our goal should be to use SDK all the time, but we are not there yet, since many users are still using YAMLs.
@StefanoFioravanzo @kubeflow/wg-training-leads What concrete steps do you suggest we can make in this PR for the short-term ?

Reply:
> I can move this to a separate guide. I think, our goal should be to use SDK all the time, but we are not there yet, since many users are still using YAMLs.

@andreyvelich I don't think so since users are not only researchers and data scientists. I can imagine there are 2 personas: researchers/data scientists and DevOps engineer/cluster admin. So, I'm curious about whether we can re-structure this section based on 2 personas.

Reply:
> I noticed that even engineers can use this SDK.

I think this is key. We don't necessarily have objective data, but I think it's pretty established that when you can, you use a programmatic SDK and a YAML spec is more like an API reference than anything else. That's why I suggest moving the YAML example to a less predominant place. It doesn't feel like a "quickstart". Remember, a quickstart is something that represents the absolute minimum for the application to show its value. The SDK is perfect for that.

Now, regarding this: 

> our goal should be to use SDK all the time, but we are not there yet, since many users are still using YAMLs.

Is this a concern about feature parity or something else? If that's not the case, then promoting the SDK as the primary way of use (starting from the quickstart) is a good way to push people to use it. We can link to the YAML how-to guide or examples at the end of the quickstart

Reply:
> That's why I suggest moving the YAML example to a less predominant place. It doesn't feel like a "quickstart". Remember, a quickstart is something that represents the absolute minimum for the application to show its value. The SDK is perfect for that.

That makes sense, it would be ideal to add support for Docker image in [`objective` parameter](https://github.com/kubeflow/katib/blob/master/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L156) in `tune` API, similar to the [`base_image` in `create_job`  to create Training Job](https://github.com/kubeflow/training-operator/blob/master/sdk/python/kubeflow/training/api/training_client.py#L327).
E.g. the experience will look like this:

```python
katib.KatibClient().tune(
    name="exp-name",
    objective="docker.io/kubeflowkatib/pytorch-mnist-cpu",
    parameters=...,
    objective_metric_name="...",
    max_trial_count=...,
    resources_per_trial={"cpu": "2"},
)
```
But we don't support it yet.

>If that's not the case, then promoting the SDK as the primary way of use (starting from the quickstart) is a good way to push people to use it. We can link to the YAML how-to guide or examples at the end of the quickstart

Sure, maybe initially I can remove this example from Getting Started page.
What place do you think we should move it ?
E.g. somewhere in the `/user-guides/hp-tuning` ?

Reply:
> But we don't support it yet.

Let's have an issue and add it to a tentative tracker for Kubeflow 1.10? :) 

> What place do you think we should move it ?
> E.g. somewhere in the /user-guides/hp-tuning ?

Sounds good. Let's not overthink it right now, any place that's good enough from your perspective. We can revisit one we merge this PR.

Reply:
Sure, created this: https://github.com/kubeflow/katib/issues/2326

Reply:
> however I noticed that even engineers can use this SDK. I noticed that for some use-cases engineers can build custom clients on top of that SDK for custom integrations.

That makes sense, but I know that Ops members prefer to use `kubectl` instead of implementing something, and they evaluate kubeflow using the `kubectl`. So, I would recommend to leave the YAML example here. However, putting deploy command should be enough, which means I think that we don't need to put YAML file here.

>  It doesn't feel like a "quickstart". Remember, a quickstart is something that represents the absolute minimum for the application to show its value.

This is the reason why I would suggest leaving the YAML example here. Because we can verify the minimum features by the single command `kubectl create -f https://raw.githubusercontent.com/kubeflow/katib/master/examples/v1beta1/hp-tuning/random.yaml`.

Reply:
I think, for OPS members it should be obvious that they can use `kubectl` to create Katib Experiments, given the nature of CRDs.
@tenzen-y from your point of view [having this example](https://deploy-preview-3723--competent-brattain-de2d6d.netlify.app/docs/components/katib/user-guides/hp-tuning/configure-experiment/#running-the-experiment) in the HP Tuning guide looks good ?

Reply:
> @tenzen-y from your point of view [having this example](https://deploy-preview-3723--competent-brattain-de2d6d.netlify.app/docs/components/katib/user-guides/hp-tuning/configure-experiment/#running-the-experiment) in the HP Tuning guide looks good ?

LGTM
It would be great to mention the section (`Katib > User Guides > Hyperparameter Tuning >How to configure Experiment > Running the Experiment `) in this `Getting Started`.

Reply:
@tenzen-y I added it to the Getting Started next steps:
https://github.com/kubeflow/website/pull/3723/commits/1435fb119dc9daa7a52baee83bac9a75168825a6

Reply:
Thank you!
/lgtm